finagle-mysql/src/main/scala/com/twitter/finagle/mysql/CanBeParameter.scala

```diff
@@ -0,0 +1,182 @@
+package com.twitter.finagle.exp.mysql
+
+import com.twitter.finagle.exp.mysql.transport.{Buffer, BufferWriter}
+
+trait CanBeParameter[-A] { outer =>
```

Comment:
@mosesn If I have a custom data type `X` such that `X <: Y`, and there's evidence `CanBeParameter[Y]` available, I would want to be able to pass `X`. The fact that it typechecks means such a usage is valid. Why not have the annotation?